cmd/global-query/pkg/distributed/querier.go

```diff
@@ -28,6 +29,14 @@ type Querier interface {
 	CreateQueryWorkload(ctx context.Context, host string, args *query.Args) (*QueryWorkload, error)
 }
 
+// QuerierAnyable extends a "common" Querier with the support to retrieve a list of all hosts / targets
+// available to the Querier
+type QuerierAnyable interface {
```

Comment:
Are we sure this is the right place for this interface? I'm all good with the proposal for an `AllHosts()` method.

IMHO, this is resolution behavior rather than query behavior. So I'd place the definition of the interface to where the `HostsResolver` interface lives, call the interface `ResolveAnyable` and have the `APIClientQuerier` simply implement the `ResolveAny` method.

Reply:
Interesting idea, thanks for the suggestion. I'll look into it!

Reply:
Mh, there's just one problem: The resolver has no way to actually _resolve_ any hosts (because the global query server uses a `StringResolverType`, and it doesn't have access to the config (which is required to grab the list of all hosts).
So yes, logically the functionality fits the `Resolver` better, but currently the resolver cannot provide the required features to actually resolve it.